main.go

```diff
@@ -118,18 +118,12 @@ func metricsHandler() http.HandlerFunc {
 				return
 			}
 		}
-
-		// Support optionally overriding collectlogs setting using a query parameter
-		collectLogs := sc.CollectLogs()
-		collectLogsOverride := r.URL.Query().Get("collectlogs")
-		if collectLogsOverride != "" {
-			if collectLogs, err = strconv.ParseBool(collectLogsOverride); err != nil {
-				targetLoggerCtx.WithError(err).Error("error parsing collectlogs query parameter as a boolean")
-				return
-			}
+		collectionCtx, err := common.NewCollectionContext(r, target, hostConfig, targetLoggerCtx)
```

Comment:
I like the addition of the context. From a quick glance, it now means that the redfish session is now long lived? I.e. exists across scrapes. In that case, we need to handle re-authentication if the session expires / is deleted etc, or have I missed something?